lib/game/client/engine/phaser/phaser-scene.driver.js

```diff
@@ -0,0 +1,191 @@
+
+class PhaserSceneDriver extends BaseSceneDriver
+{
+    constructor(config)
+    {
+        super();
+        this.driverName = "Phaser Scene Driver";
+        this.scene = new WrapperScene(config);
+    }
+
+    setUpPreloadCallback(callback)
+    {
+        this.getScene().setUpPreloadCallback(callback);
+    }
+
+    setUpCreateCallback(callback)
+    {
+        this.getScene().setUpCreateCallback(callback);
+    }
+
+    loadHTML(key, url, xhrSettings)
+    {
+        this.getScene().load.html(key, url, xhrSettings);
+    }
+
+    loadTilemapTiledJSON(key, url, xhrSettings)
+    {
+        this.getScene().load.tilemapTiledJSON(key, url, xhrSettings);
+    }
+
+    loadSpritesheet(key, url, frameConfig, xhrSettings)
+    {
+        this.getScene().load.spritesheet(key, url, frameConfig, xhrSettings);
+    }
+
+    loadImage(key, url, xhrSettings)
+    {
+        this.getScene().load.image(key, url, xhrSettings);
+    }
+
+    loadOn(event, fn, context)
+    {
+        this.getScene().load.on(event, fn, context);
+    }
+
+    getCameraManager()
+    {
+        return this.getScene().cameras;
+    }
+
+    getMainCamera()
+    {
+        return this.getCameraManager().main;
+    }
+
+    getMainCameraWidth()
+    {
+        return this.getMainCamera().width;
+    }
+
+    getMainCameraHeight()
+    {
+        return this.getMainCamera().height;
+    }
+
+    addGraphics()
+    {
+        return this.getScene().add.graphics();
+    }
+
+    addDomCreateFromCache(x, y, config)
+    {
+        return this.getScene().add.dom(x, y, config.element, config.style, config.innerText).createFromCache(config.the);
+    }
+
+    getCacheHtml(key)
+    {
+        return this.getScene().cache.html.get(key);
+    }
+
+    createAnimation(config)
+    {
+        return this.getScene().anims.create(config);
+    }
+
+    generateAnimationFrameNumbers(key, config)
+    {
+        return this.getScene().anims.generateFrameNumbers(key, config);
+    }
+
+    addText(x, y, text, style)
+    {
+        return this.getScene().add.text(x, y, text, style);
+    }
+
+    getRectangle()
+    {
+        return Geom.Rectangle;
+    }
+
+    getScenePlugin()
+    {
+        return this.getScene().scene;
+    }
+
+    getChildren()
+    {
+        return this.getScene().children;
+    }
+
+    inputKeyboardOn(event, key)
+    {
+        return this.getScene().input.keyboard.on(event, key);
+    }
+
+    setUiElement(uiName, value)
+    {
+        this.getScene().setUiElement(uiName, value);
+    }
+
+    getUiElement(uiName, logError = true)
+    {
+        return this.getScene().getUiElement(uiName, logError);
+    }
+
+    getAllUiElements()
+    {
+        return this.getScene().getAllUiElements();
+    }
+
+    getScene()
+    {
+        return this.scene;
+    }
+}
+
+class WrapperScene extends Scene
+{
+
+    constructor(config)
+    {
+        super(config);
+        this.elementsUi = {};
+    }
+
+    setUpPreloadCallback(callback)
```

Comment:
as general code style note, even if you see some old code with setters and getters avoid these by any means

Reply:
I should pass everything as a constructor parameter when it can be done that way.

Reply:
I was refering more to, don't write unnecesary methods, for example:
`this.getScene()` when you can actually just: `this.scene`

Reply:
Done!